### DISCUSSION TITLE
Add required CSS custom properties with utilities

### DISCUSSION THREAD

**Discussion by jacobrask** [tailwindlabs/tailwindcss#11393] [Ideas]
If I want to use Tailwind without the included base layer I won't get all the CSS custom properties required for a utility class. Utility classes such as `ring` rely on properties such as `--tw-ring-inset` which are set in the base layer.

Example: https://play.tailwindcss.com/dElDtb8cTR?file=css

I'd expect either a separate layer for all required custom properties - or ideally that if a utility class is included, it's required custom property is added, but not otherwise.

**Comment by wongjn** [comment]
You could consider keeping `@tailwind base` but [disabling the core Tailwind preflight plugin](https://tailwindcss.com/docs/preflight#disabling-preflight) if are removing `@tailwind base` only to remove the Tailwind base styling: https://play.tailwindcss.com/EKflV8FOlv?file=config
